GliaWidgets/SecureConversations/SecureConversations.SendMessageButton.swift

```diff
@@ -0,0 +1,158 @@
+        }
+
+        func setup() {
+            addSubview(messageTitleStackView)
+            NSLayoutConstraint.activate([
+                messageTitleStackView.centerXAnchor.constraint(equalTo: centerXAnchor),
+                messageTitleStackView.centerYAnchor.constraint(equalTo: centerYAnchor)
+            ])
+        }
+
+        func renderProps() {
+            let props = UIKitProps(props)
+            messageTitleLabel.text = props.title
+            messageTitleLabel.font = props.titleFont
+            messageTitleLabel.textColor = props.forgroundColor
+            backgroundColor = props.backgroundColor
+            layer.borderWidth = props.borderWidth
+            layer.borderColor = props.borderColor.cgColor
+            layer.cornerRadius = props.cornerRadius
+            activityIndicator.color = props.acitivityIndicatorColor
+            renderedIsIndicatorShown = props.isActivityIndicatorShown
+        }
+
+        var renderedIsIndicatorShown: Bool = false {
+            didSet {
+                guard renderedIsIndicatorShown != oldValue else { return }
+                activityIndicator.isHidden = !renderedIsIndicatorShown
+
+                if renderedIsIndicatorShown {
+                    activityIndicator.startAnimating()
+                } else {
+                    activityIndicator.stopAnimating()
+                }
+            }
+        }
+    }
+}
+
+extension SecureConversations.SendMessageButton {
+    enum Props: Equatable {
+        case normal(NormalState)
+        case disabled(DisabledState)
+    }
+}
+
+extension SecureConversations.SendMessageButton.Props {
+    struct NormalState: Equatable {
+        let title: String
+        let titleFont: UIFont
+        let forgroundColor: UIColor
+        let backgroundColor: UIColor
+        let borderColor: UIColor
+        let borderWidth: Double
+        let cornerRadius: Double
+        let acitivityIndicatorColor: UIColor
+        let isActivityIndicatorShown: Bool
+    }
+
+    struct DisabledState: Equatable {
+        let title: String
+        let titleFont: UIFont
+        let forgroundColor: UIColor
+        let backgroundColor: UIColor
+        let borderColor: UIColor
+        let borderWidth: Double
+        let cornerRadius: Double
+        let acitivityIndicatorColor: UIColor
+        let isActivityIndicatorShown: Bool
+    }
+}
+
+extension SecureConversations.SendMessageButton {
+    struct UIKitProps: Equatable {
+        let title: String
+        let titleFont: UIFont
+        let forgroundColor: UIColor
+        let backgroundColor: UIColor
+        let borderColor: UIColor
+        let borderWidth: Double
+        let cornerRadius: Double
+        let acitivityIndicatorColor: UIColor
+        let isActivityIndicatorShown: Bool
+        let isEnabled: Bool
+
+        init(_ props: Props) {
+            switch props {
+            case let .normal(state):
+                self.title = state.title
```

Comment:
Is there a way to optimize this? I don't know because you will probably not be able to send these calls to another method, because both states are the same except for the `isEnabled`.

Reply:
There computed property `isEnabled` can be added to `Props`

Reply:
States passed from props have different values. The point of UIKitProps is to use exhaustive switch and assign correct values, so I'm not sure how to do it differently.

Reply:
@EgorovEI I think it is not worth the effort in this particular case.